app/scripts/lib/rpc-method-middleware/createMethodMiddleware.js

```diff
@@ -0,0 +1,35 @@
+import handlers from './handlers'
+
+/**
+ * The purpose of this middleware is to create portable RPC method
+ * implementations that are decoupled from the rest of our background
+ * architecture.
+ * 
+ * Handlers consume functions that hook into the background, and only depend
+ * on their signatures, not e.g. controller internals.
+ * 
+ * Eventually, we'll want to extract this middleware into its own package.
+ */
+
+const handlerMap = handlers.reduce((map, handler) => {
+  map.set(handler.methodName, handler.implementation)
+  return map
+}, new Map())
+
+/**
+ * Returns a middleware that implements the following RPC methods:
+ * - metamask_logInjectedWeb3Usage
```

Comment:
Technically only if 'handlers' includes 'log-web3-usage' in it, right? This might confuse. I think the docblock on handlerMap covers the variadic nature of the handlers...